src/syft/core/pointer/pointer.py

```diff
@@ -145,22 +146,40 @@ def __init__(
 
     def get(
         self,
+        request_block: bool = False,
     ) -> StorableObject:
         """Method to download a remote object from a pointer object if you have the right
         permissions.
 
         :return: returns the downloaded data
         :rtype: StorableObject
         """
-        obj_msg = GetObjectAction(
-            obj_id=self.id_at_location,
-            address=self.client.address,
-            reply_to=self.client.address,
-        )
 
-        response = self.client.send_immediate_msg_with_reply(msg=obj_msg)
+        if request_block:
+            from ..node.domain.service import RequestStatus
+
+            response_status = self.request(block=True)
+
+            if response_status == RequestStatus.Accepted:
+                obj_msg = GetObjectAction(
+                    obj_id=self.id_at_location,
+                    address=self.client.address,
+                    reply_to=self.client.address,
+                )
+
+                response = self.client.send_immediate_msg_with_reply(msg=obj_msg)
+
+                return response.obj
+        else:
```

Comment:
This and the above block of code can be improved [Need Suggestions}

Reply:
Maybe have an internal function that simply doe:
```
obj_msg = GetObjectAction(
                    obj_id=self.id_at_location,
                    address=self.client.address,
                    reply_to=self.client.address,
                )

                response = self.client.send_immediate_msg_with_reply(msg=obj_msg)

                return response.obj
```